tasks/test.rake

```diff
@@ -13,31 +15,27 @@ end
 desc "Run the full suite using 1 core"
 task test: ['spec:unit', 'spec:request', 'cucumber', 'cucumber:class_reloading']
 
-require 'coveralls/rake/task'
-Coveralls::RakeTask.new
-task test_with_coveralls: [:test, 'coveralls:push']
+unless ENV["NOCOVER"]
+  require 'coveralls/rake/task'
+  Coveralls::RakeTask.new
+  task test_with_coveralls: [:test, 'coveralls:push']
+end
 
 namespace :test do
 
   def run_tests_against(*versions)
-    current_version = detect_rails_version if File.exists?("Gemfile.lock")
-
-    versions.each do |version|
+    versions.each do |fuzzy_rails_version|
       puts
-      puts "== Using Rails #{version}"
-
-      cmd "./script/use_rails #{version}"
-      cmd "bundle exec rspec spec"
-      cmd "bundle exec cucumber features"
-      cmd "bundle exec cucumber -p class-reloading features"
+      puts "== Using Rails #{fuzzy_rails_version}"
+      cmd "BUNDLE_GEMFILE='gemfiles/Gemfile.rails-#{fuzzy_rails_version}' bundle exec rspec spec"
+      cmd "BUNDLE_GEMFILE='gemfiles/Gemfile.rails-#{fuzzy_rails_version}' bundle exec cucumber features"
+      cmd "BUNDLE_GEMFILE='gemfiles/Gemfile.rails-#{fuzzy_rails_version}' bundle exec cucumber -p class-reloading features"
     end
-
-    cmd "./script/use_rails #{current_version}" if current_version
   end
 
   desc "Run the full suite against the important versions of rails"
   task :major_supported_rails do
-    run_tests_against *TRAVIS_RAILS_VERSIONS
+    run_tests_against *%(3.2.x 4.1.x 4.2.x)
```

Comment:
Definitely.  I question is we keep this whole `namespace :test do` section though.  So much of this manual code will be obviated by Appraisal (not yet in the PR, but will be soon).